### PR TITLE
Custom metrics formatting in tqdm status bar

### PR DESCRIPTION
**Pull request by lminer**
This is a simple change that will allow custom formatting of metrics for the tqdm status bar. This is especially useful if you have small loss values and want to see it in scientific notation. Replace `metrics_format` with `'{name}: {value:e}'`

### CODE REVIEW

**Comment by seanpmorgan** [comment]
cc @shun-lin

**Comment by shun-lin** [comment]
Looks good to me :)
Thanks for the contribution!

**Comment by lminer** [comment]
@googlebot I signed it!

**Comment by lminer** [comment]
@seanpmorgan is this good to merge?

**Comment by casperdcl** [comment]
Ah I never saw this earlier. Looks like it fixes #708.

Anyone looked at https://github.com/tqdm/tqdm/#keras-integration? In particular https://github.com/tqdm/tqdm/blob/master/tqdm/keras.py - looks like it was developed at the same time as the tensorflow addon. Seems silly to have it in two places.

I'd suggest opening a PR on https://github.com/tqdm/tqdm/ to change anything that may be required and then changing https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/tqdm_progress_bar.py to just import from tqdm...

cc @shun-lin @gabrieldemarmiesse @seanpmorgan

**Comment by shun-lin** [comment]
Hi @casperdcl,

Thanks for the suggestion! I think this PR does not require any change in the tqdm repo as it deals with how users want to display the metrics return from the training loops. `tqdm_progress_bar.py` does indeed import `tqdm` (specifically `tqdm.auto`, we put the import in the initialization of the class because `tqdm` is not a required package for Tensorflow Addons). The TQDM progress bar here is compatible with `tensorflow.keras` and not `keras` with other backends (we import `tensorflow.keras.Callback` instead of `keras.Callback` but it may work with other backends if you simply change that line to import from `keras.Callback` but I have not tried it out yet).

https://github.com/tensorflow/addons/blob/1bbe67ff075040eb54a468a3993dd36a0a5689c3/tensorflow_addons/callbacks/tqdm_progress_bar.py#L60-L71

Here is the link to the tutorial:
https://github.com/tensorflow/addons/blob/master/docs/tutorials/tqdm_progress_bar.ipynb

Thanks! :)

**Comment by casperdcl** [comment]
@shun-lin I understand but I was referring to future development/PRs to ease the burden of maintenance. Right now there's `tf.addons.callbacks.tqdm_progress_bar` being developed in tandem with `tqdm.keras` even though they are broadly identical. I was suggesting a way to avoid code/logic duplication between them.

**Comment by shun-lin** [comment]
> @shun-lin I understand but I was referring to future development/PRs to ease the burden of maintenance. Right now there's `tf.addons.callbacks.tqdm_progress_bar` being developed in tandem with `tqdm.keras` even though they are broadly identical. I was suggesting a way to avoid code/logic duplication between them.

oh got it thanks for clarifying, and yes I completely agree with you on that we should find a way to deduplicate code!
